graylog2-shared/src/main/java/org/graylog2/shared/buffers/processors/ProcessBufferProcessor.java

```diff
@@ -55,25 +57,29 @@ public void onEvent(MessageEvent event) throws Exception {
         // TODO The DecodingProcessor does not need to be a EventHandler. We decided to do it like this to keep the change as small as possible for 1.0.0.
         decodingProcessor.onEvent(event, 0L, false);
 
-        final Message msg = event.getMessage();
-        if (msg == null) {
+        final Collection<Message> messageList = event.getMessages();
+        if (messageList == null) {
             // skip message events which could not be decoded properly
             return;
         }
-        incomingMessages.mark();
-        final Timer.Context tcx = processTime.time();
 
+        for (final Message message : messageList) {
```

Comment:
this would simply change into `if (event.getMsg() == null) { for ...} else {dispatchMessage(event.getMsg())}`, right?